clash-lib/src/Clash/Util.hs

```diff
@@ -295,24 +254,6 @@ indexNote
   -- ^ Error or element /n/
 indexNote note = \xs i -> fromMaybe (error note) (indexMaybe xs i)
 
--- | Safe version of 'head'
-headMaybe :: [a] -> Maybe a
-headMaybe (a:_) = Just a
-headMaybe _ = Nothing
-
--- | Safe version of 'tail'
-tailMaybe :: [a] -> Maybe [a]
-tailMaybe (_:as) = Just as
-tailMaybe _ = Nothing
-
--- | Split the second list at the length of the first list
-splitAtList :: [b] -> [a] -> ([a], [a])
```

Comment:
Why remove this? `splitAtList xs ys` is faster than `splitAt (lenght xs) ys` when `xs` is bigger than `ys`, as you don't have to fully traverse `xs`